### PR TITLE
Remove the need to specify api/ files, pull directly from the DCL

### PR DESCRIPTION
**Pull request by rileykarson**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/8201

This PR removes the need to specify `api/` files for resources (although not samples quite yet) and pulls them directly off of the DCL's exported YAMLs. This means we'll stay up-to-date for better or for worse, although generally for the better. There's some pain in the upgrade- our YAMLs had drifted, and there were some dangerous-to-apply updates last I checked.

Changes are contained to `main.go`, and all other changes in the PR are non-functional (removing api files, removing old overrides)

Marked as WIP because of the drift- we need to resolve those to compatible before we can merge this. Outstanding issues:

- [x] CloudBuild WorkerPool bad name description: b/200841783
- [x] CloudBuild WorkerPool update: b/199915504
- [x] CloudBuild WorkerPool WorkerConfig O+C: cl/399473374
- [x] Dataproc WorkflowTemplate: b/199937950
- [x] Dataproc WorkflowTemplate version: https://github.com/GoogleCloudPlatform/magic-modules/pull/5249
- [x] EventArc Trigger descriptions: cl/396655269
- [x] GKEHub Feature `config_membership` going required: Not believed to be a problem, as the field is effectively required already
- [x] GKEHub FeatureMembership typo: Fixed upstream, will be picked up after merging with https://github.com/GoogleCloudPlatform/magic-modules/pull/5201
- [x] GKEHub FeatureMembership `configmanagement`/`location`/`feature`/`membership` going required: Not believed to be a problem, same as for `Feature`
- [x] PrivateCA CertificateTemplate losing state hint: Spoke with Nat, not believed to be a problem

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:breaking-change
gke_hub: made the `config_membership` field in `google_gke_hub_feature` required, disallowing invalid configurations
```

```release-note:breaking-change
gke_hub: made the `configmanagement`, `feature`, `location`, `membership` fields in `google_gke_hub_feature_membership` required, disallowing invalid configurations
```

```release-note:enhancement
eventarc: added support for `uid` output field, `cloud_function` destination to `google_eventarc_trigger`
```

```release-note:enhancement
gke_hub: added support for `resource_state`, `state` outputs to `google_gke_hub_feature`
```

```release-note:enhancement
gke_hub: added support for `gcp_service_account_email` when configuring Git sync in `google_gke_hub_feature_membership`
```

### CODE REVIEW

**Comment by rileykarson** [comment]
Actually, wait, found a couple more cleanups I wanted to make.

**Comment by rileykarson** [comment]
Outstanding issues:

* CloudBuild WorkerPool update: b/199915504
* Dataproc WorkflowTemplate: Chat thread
* EventArc Trigger descriptions: cl/396655269
* GKEHub Feature `config_membership` going required: Not believed to be a problem, as the field is effectively required already
* GKEHub FeatureMembership typo: Fixed upstream, will be picked up after merging with https://github.com/GoogleCloudPlatform/magic-modules/pull/5201
* GKEHub FeatureMembership `configmanagement`/`location`/`feature`/`membership` going required: Not believed to be a problem
* PrivateCA CertificateTemplate losing state hint: Spoke with Nat, not believed to be a problem

File: tpgtools/serialization.go
Comment (slevenick) [inline tpgtools/serialization.go:1740]: Meta, but do we have a way to either untrack this file or make diffs not show up in GitHub? It being auto-generated within the MM repo causes extra noise

**Review by slevenick — COMMENTED** [review]
Looks fine to me once all the DCL resources are fixed!

File: tpgtools/serialization.go
Comment (rileykarson) [reply tpgtools/serialization.go:1740]: Filed https://github.com/hashicorp/terraform-provider-google/issues/10170, I know @ndmckinley had some thoughts about improving our use of `go generate`.

**Comment by rileykarson** [comment]
@slevenick any concern if I squash down to one commit when I resolve those merge conflicts? Tracking `go.mod` changes across a few different commits is annoying to rebase.

**Comment by slevenick** [comment]
> @slevenick any concern if I squash down to one commit when I resolve those merge conflicts? Tracking `go.mod` changes across a few different commits is annoying to rebase.

No concerns! It happens on merge anyways so 🤷

**Comment by rileykarson** [comment]
Third try's the charm! /gcbrun

**Review by slevenick — APPROVED** [review]
LGTM assuming tests pass!

**Comment by rileykarson** [comment]
Weird- VCR ran, the Magician didn't post a message on the initial run, and it did on the recording. I confirmed the build #s corresponded to the right TPGB commit, too!

**Pull request merged**
